src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityDataHatch.java

```diff
@@ -0,0 +1,93 @@
+package gregtech.common.metatileentities.multi.multiblockpart;
+
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.capability.impl.NotifiableItemStackHandler;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockAbilityPart;
+import gregtech.api.metatileentity.multiblock.MultiblockAbility;
+import gregtech.client.renderer.texture.Textures;
+import gregtech.client.renderer.texture.cube.SimpleOverlayRenderer;
+import gregtech.client.utils.PipelineUtil;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+import java.util.List;
+
+public class MetaTileEntityDataHatch extends MetaTileEntityMultiblockNotifiablePart implements IMultiblockAbilityPart<IItemHandlerModifiable> {
+    private final NotifiableItemStackHandler dataStickInventory;
+    private final int slotAmount;
+
+    public MetaTileEntityDataHatch(ResourceLocation metaTileEntityId, int tier) {
+        super(metaTileEntityId, tier, false);
+        int[] invSizes = {1, 1, 1, 1, 4, 8, 16, 32, 64};
+        this.slotAmount = invSizes[tier];
+        this.dataStickInventory = new NotifiableItemStackHandler(slotAmount+1, getController(), false);
```

Comment:
Why is this a +1? Additionally, this should be done in the `createImportItemHandler()` method, and an export inventory in the `createExportItemHandler()` method. Look at the item bus for examples.

Reply:
some testing code left over